hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/service/mono/txns/contract/ContractCreateTransitionLogic.java

```diff
@@ -210,9 +210,6 @@ public void doStateTransitionOperation(
                         codeWithConstructorArgs,
                         consensusTime);
             } else {
-                sender.incrementEthereumNonce();
-                accountStore.commitAccount(sender);
-
                 result = evmTxProcessor.executeEth(
```

Comment:
Why don't we set the result's `signerNonce` following this `executeEth()` call (in analogy to `ContractCallTransitionLogic`)?

Reply:
Good catch, thank you! Fixed it and also added a test for ethereum contract deployment.

I haven't added in `EvmTxProcessor.executeEth()`, because there are many more places that have to be handled. E.g. both `ContractCreateTransitionLogic` and `ContractCallTransitionLogic` are leading to `CreateEvmTxProcessor.executeEth()` and `CallEvmProcessor.executeEth()`. And again, they both lead to `EvmTxProcessor.execute()` where I can potentially add the `signer_nonce` to the `TransactionProcessingResult`, but there are several different `return` places that must be handled. So, I took the path that was to add them, once the result had been already built. If you think that the better approach is to move `result.setSignerNonce()` to `EvmTxProcessor`, I'll refactor the current implementation :pray:.

Reply:
Ah np, this looks fine to me. 🙏